### PR TITLE
Track analytics for start time, channel and team switching

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Start tracking the following times in **ms** and send them to segment
* Start time (the amount of time it takes to first load the channel once logged in)
* Channel switch (the amount of time it takes to switch channels)
* Team switch (the amount of time it takes to switch team)

#### Ticket Link
https://mattermost.atlassian.net/browse/RN-533

#### Screenshots
[If the PR includes UI changes, include screenshots (for both iOS and Android if possible).]
image

### CODE REVIEW

**Comment by hmhealey** [comment]
Code looks good, but do we have docs on what we track here?

**Comment by enahum** [comment]
@hmhealey @mkraft feel free to review this one, I'll rebase it

File: app/utils/segment.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            userId: diagnosticId,
+            context: global.analytics_context,
+            page: {
+                path: '',
+                referrer: '',
+                search: '',
+                title: '',
+                url: ''
+            },
+            anonymousId: '00000000000000000000000000'
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/utils/segment.js]: not a mistake

**Comment by enahum** [comment]
@mkraft changes made

**Pull request merged**